File: usaco/usaco guide/bronze/simulation/circularbarn.cpp
#include <bits/stdc++.h>
#include <iostream>
using namespace std;

int main() {
    freopen("cbarn.in", "r", stdin);
    freopen("cbarn.out", "w", stdout);
    int N; cin >> N;
    int arr[N];
    for (int i = 0; i<N; i++) {
        int temp; cin >> temp;
        arr[i] = temp;
    }

    int res = 999999999;
    for (int i = 0; i<N; i++) {
        int temp = 0;
        for (int j = 0; j<N; j++) {
            if (j < i) {
                temp += (arr[j] * (N - (i-j)));
            }
            else if (i != j) {
                temp += (arr[j] * (j-i));
            }
        }
        res = min(res, temp);
    }
    cout << res;
    return 0;
}

File: usaco/usaco guide/bronze/simulation/circularbarn.py
import sys

sys.stdin = open("cbarn.in", "r")
sys.stdout = open("cbarn.out", "w")

def solve():
    n = int(input())
    arr = []
    for i in range(n):
        arr.append(int(input()))
        
    res = float('inf')
    for i in range(len(arr)):
        temp = 0
        for j in range(len(arr)):
            if j < i:
                temp += (arr[j] * (len(arr) - (i-j)))
            elif i != j:
                temp += (arr[j] * (j-i))
        res = min(res, temp)
    return res

print(solve())